File: rest-common/src/main/java/org/chronopolis/rest/models/storage/FixityCreate.java
package org.chronopolis.rest.models.storage;

public class FixityCreate {

    private String algorithm;
    private String value;

    public String getAlgorithm() {
        return algorithm;
    }

    public FixityCreate setAlgorithm(String algorithm) {
        this.algorithm = algorithm;
        return this;
    }

    public String getValue() {
        return value;
    }

    public FixityCreate setValue(String value) {
        this.value = value;
        return this;
    }
}

File: rest-models/src/main/kotlin/org/chronopolis/rest/models/create/FixityCreate.kt
package org.chronopolis.rest.models.create

import org.chronopolis.rest.models.Fixity
import org.chronopolis.rest.models.enums.FixityAlgorithm
import javax.validation.constraints.NotEmpty

/**
 * Request to create a [Fixity]
 *
 * @property algorithm the algorithm used
 * @property value the value the algorithm produced
 * @author shake
 */
data class FixityCreate(var algorithm: FixityAlgorithm = FixityAlgorithm.SHA_256,
                        @get:NotEmpty var value: String = "")
